src/screen.cpp

```diff
@@ -117,10 +117,11 @@ static void _screen_error(int fatal, bool use_errno, int error, const char *fmt,
 
     GET_TIME (&currentTime);
 
-    c+= sprintf(c, "%s: ", CStat::formatTime(&currentTime));
-    c+= vsprintf(c, fmt, ap);
+    const std::size_t bufSize = sizeof(screen_last_error)/sizeof(screen_last_error[0]);
+    c+= snprintf(c, bufSize, "%s: ", CStat::formatTime(&currentTime));
+    c+= vsnprintf(c, (bufSize - (c - screen_last_error)), fmt, ap);
```

Comment:
Please add spaces around `/` and `+=`.

(When you're touching lines, might as well fix the code style.)

And, you cannot add the return value of (v)snprintf() to c, because:
> If the output was truncated due to this limit, then the return value is the number of characters (excluding the terminating null byte) which would have been written to the final string if enough space had been available.

So that'll need fixing.

Reply:
Good catch on the return value. Turns out I also need to check for a negative return value. I've just pushed a new version of the fix. Thanks.

Reply:
As far as I can tell, the changes are working well. I was able to do a fair amount of testing.